sdk/python/approzium/opentelemetry/__init__.py

```diff
@@ -0,0 +1,51 @@
+import opentelemetry.ext.dbapi
+from opentelemetry.ext.dbapi import DatabaseApiIntegration
+
+original_DatabaseApiIntegration = None
+
+
+class _ApproziumDatabaseApiIntegration(DatabaseApiIntegration):
+    def get_connection_attributes(self, connection):
+        super().get_connection_attributes(connection)
+        # check if this is an Approzium connection
+        if hasattr(connection, "authenticator"):
+            self.add_approzium_span_attributes(connection)
+
+    def add_approzium_span_attributes(self, connection):
+        for key, value in connection.authenticator.attribution_info.items():
+            self.span_attributes["approzium." + key] = value
+
+
+def instrument():
+    """Instruments Opentelemetry ``opentelemetry.ext.dbapi`` to automatically include
+    attribution tags in traces generated by Approzium database connections. Call this
+    method before calling the ``instrument`` method provided by Opentelemetry database
+    extensions.
+
+    Refer to :py:obj:`approzium.AuthClient.attribution_info` for generated tags.
+
+    Example (using Opentelemetry Psycopg2 integration):
+
+    .. code-block:: python
+
+        >>> import approzium.opentelemetry
+        >>> from approzium.psycopg2 import connect
+        >>> from opentelemetry.ext.psycopg2 import Psycopg2Instrumentor
+        >>> approzium.opentelemetry.instrument()
+        >>> Psycopg2Instrumentor().instrument()
+        >>> cnx = connect("host=dbmd5 dbname=db user=bob")
+        >>> cursor.execute("SELECT 1")
+    """
+    global original_DatabaseApiIntegration
+    original_DatabaseApiIntegration = DatabaseApiIntegration
+    opentelemetry.ext.dbapi.DatabaseApiIntegration = _ApproziumDatabaseApiIntegration
+
+
+def uninstrument():
```

Comment:
Is this something that must be called if you call instrument? If so, can we add a comment to that effect?

Reply:
Not necessarily! You can call instrument and forget about it, just like the opentelemetry interface. If you wish to uninstrument, they give that option and so I give that too.

Reply:
Would you mind adding a comment to that effect? Because I bet others will wonder too. And/or adding it to an example would likewise help.